reference/constraints/_payload-option.rst.inc

```diff
@@ -0,0 +1,6 @@
+payload
+~~~~~~~
+
+**type**: ``mixed`` **default**: ``null``
+
+This option can be used to attach arbitrary domain-specific data to a constraint.
```

Comment:
Not sure if that is really needed. I you don't know how to use this, you will probably don't have the need for it or do you have a use case in mind that deserves to be explained here?